io/EptReader.cpp

```diff
@@ -241,23 +265,45 @@ void EptReader::addDimensions(PointLayoutPtr layout)
     const Json::Value& schema(m_info->schema());
     m_remoteLayout.reset(new FixedPointLayout());
 
-    // Register XYZ directly since they may appear as scaled int32 values in the
-    // schema.  Either way we want doubles.
-    layout->registerDim(Dimension::Id::X);
-    layout->registerDim(Dimension::Id::Y);
-    layout->registerDim(Dimension::Id::Z);
-
     for (Json::ArrayIndex i(0); i < schema.size(); ++i)
     {
         const Json::Value& dim(schema[i]);
         const std::string name(dim["name"].asString());
-        log()->get(LogLevel::Debug) << "Registering dim " << name << std::endl;
-        const Dimension::Type type(Dimension::type(dim["type"].asString()));
+
+        // If the dimension has a scale, make sure we register it as a double
+        // rather than its serialized type.
+        const Dimension::Type type = getType(dim);
+
+        log()->get(LogLevel::Debug) << "Registering dim " << name << ": " <<
+            Dimension::interpretationName(type) << std::endl;
+
         layout->registerOrAssignDim(name, type);
         m_remoteLayout->registerOrAssignDim(name, type);
     }
 
     m_remoteLayout->finalize();
+
+    using D = Dimension::Id;
+
+    m_dimTypes = m_remoteLayout->dimTypes();
+    for (DimType& dt : m_dimTypes)
+    {
+        // If the data is stored as Laszip, then PDAL's LasReader will unscale
+        // XYZ for us.
+        if (m_info->dataType() == EptInfo::DataType::Laszip &&
+                (dt.m_id == D::X || dt.m_id == D::Y || dt.m_id == D::Z))
+        {
+            continue;
+        }
+
+        const Json::Value dim(m_info->dim(m_remoteLayout->dimName(dt.m_id)));
+        if (dim.isNull())
```

Comment:
There's a bunch of code in EptInfo to create m_transforms and then it's not used.  There's this stuff instead.

Reply:
Fixed in 5a752aadedd66938174b734acde7c1a1c884087b.